### PR TITLE
change - handle market type and params

### PR DESCRIPTION
**Pull request by ttodua**
I think this should be the correct way.

### CODE REVIEW

File: python/ccxt/base/exchange.py [hunk 1]
```diff
@@ -2769,7 +2769,7 @@ def handle_market_type_and_params(self, method_name, market=None, params={}):
             if isinstance(method_options, str):
                 method_type = method_options
             else:
-                method_type = self.safe_string_2(method_options, 'defaultType', 'type')
+                method_type = self.safe_string_2(method_options, 'defaultType', 'type', default_type)
```
Comment (frosty00) [inline python/ccxt/base/exchange.py:2772]: can we make the python the same as the other langs

Comment (ttodua) [reply python/ccxt/base/exchange.py]: ouch, mechanical mistake of mine.

**Comment by kroitor** [comment]
I don't think that we should merge this PR. The method override should not be mixed with the market type.

**Comment by kroitor** [comment]
Actually, on a second thought, I may be wrong, will review again shortly.

**Comment by ttodua** [comment]
Hm, on a second thought, i think this might be a bit more problematic.
for example, as you see implementation of that function, the `method_options` take superiority. so, even if ccxt user sets the `exchange.defaultType = 'swap'` (or whatever) , then still, if under `options` there is `fetchMarkets: { defaultType: 'spot' }` then the exchange will still load `spot` markets, even though user think he is loading swap.

I think, the main `exchange.defaultType` should be the only one user should care of. most of users cant manually go through each individual function options to check/find-out whether they are defaulted to other than `exchange.defaultType`. I think, this is a problem and extra-fuss for users.

**Comment by kroitor** [comment]
@ttodua the per-method options (like options['fetchMarkets']['defaultType/type'] take precedence over the exchange-wide options (`options['defaultType']`). The market induced from the symbol takes precedence over the options and types, since we can derive the market type from the market itself. The `params` overrides take precedence over both the market and the options. So i think the existing code is ok in regard of that. Will merge this PR shortly.

**Comment by ttodua** [comment]
> @ttodua the per-method options ...
yes, but I am thinking, what are valid cases (at all, why should there be any need to set specific method to specific `defaultType` under `options`). so, given my above argument, when user sets i.e. `exch.defaultType = 'swap'` and that exchange has `options['fetchOrders']['type'] ='spot'`  (or any fetch method, that is not 'market-specific') then the user will think he is getting data from swap, but actually from spot.  I thought about that. otherwise, your reasoning above is right, for 'market' related fetch functions, the market-type will handle itself, and if the rest cases aren't problem, then good.

**Pull request merged**